lib/ansible/modules/windows/win_updates.ps1

```diff
@@ -184,11 +187,28 @@ $update_script_block = {
                 Write-DebugLog -msg "Skipping hidden update $($update_info.title)"
                 continue
             }
-        
-            Write-DebugLog -msg "Adding update $($update_info.id) - $($update_info.title)"
-            $updates_to_install.Add($update) > $null
-        
-            $result.updates[$update_info.id] = $update_info
+
+            $post_cat_matched = $False
+            If ($post_categories.Count -eq 0) {
+                # There are no additional categories to check
+                $post_cat_matched = $True
+            }
+
+            ForEach ($update_cat in $update.Categories) {
+                $update_info.categories.Add($update_cat.Name) | Out-Null
+                ForEach ($match_cat in $post_categories) {
+                    If ($update_cat.Name -Match $match_cat) {
+                        $post_cat_matched = $True
```

Comment:
Once we get a match, best to break the loop, no need to waste cycles if we already know there's a match.

Reply:
Adjusted in next commit